src/parser.c

```diff
@@ -397,17 +397,23 @@ get_json_as_hash(char *json, int len, const char *funcname)
     state->lex           = lex;
 
     sem->semstate           = (void *) state;
-    sem->array_start        = hash_array_start;
-    sem->scalar             = hash_scalar;
-    sem->object_field_start = hash_object_field_start;
-    sem->object_field_end   = hash_object_field_end;
+    sem->array_start        = (void *) hash_array_start;
+    sem->scalar             = (void *) hash_scalar;
+    sem->object_field_start = (void *) hash_object_field_start;
+    sem->object_field_end   = (void *) hash_object_field_end;
```

Comment:
It seems that since Postgres 16 these callbacks should return `JsonParseErrorType`:
https://github.com/postgres/postgres/blob/REL_16_2/src/include/common/jsonapi.h#L89-L92

So it would be good to not ignore it. It will be necessary to modify `hash_array_start`, `hash_scalar`, `hash_object_field_start`, `hash_object_field_start`, `hash_object_field_end` to return `JSON_SUCCESS` in case of Postgres16.

Here is an example:
https://github.com/postgres/postgres/blob/REL_16_2/src/backend/utils/adt/jsonfuncs.c#L1163